src/hotspot/share/ci/ciObject.hpp

```diff
@@ -57,6 +59,24 @@ class ciObject : public ciBaseObject {
   jobject  _handle;
   ciKlass* _klass;
 
+  // Cache constant value lookups to ensure that consistent values are observed during compilation.
+  class ConstantValue {
+    private:
+      const ciObject* _obj;
```

Comment:
Since `_constant_values` is instance-specific, there's no need in `_obj` anymore, right?

Reply:
Good catch, removed.